kettle-plugins/mqtt/src/main/java/org/pentaho/di/trans/step/mqtt/MQTTConsumer.java

```diff
@@ -65,7 +68,14 @@ public boolean init( StepMetaInterface stepMetaInterface, StepDataInterface step
       logger.error( getString( PKG, "MQTTInput.Error.FailureGettingFields" ), e );
     }
     window = new FixedTimeStreamWindow<>( subtransExecutor, rowMeta, getDuration(), getBatchSize() );
-    source = new MQTTStreamSource( mqttConsumerMeta.getMqttServer(), mqttConsumerMeta.getTopics(), 2, this );
+    try {
+      source = new MQTTStreamSource( environmentSubstitute( mqttConsumerMeta.getMqttServer() ),
+        Arrays.asList( environmentSubstitute( mqttConsumerMeta.getTopics().toArray( new String[]{} ) ) ),
```

Comment:
should give it an array of the exact size

Reply:
👍